docs/conf.py

```diff
@@ -201,3 +196,11 @@
 
 # A list of files that should not be packed into the epub file.
 epub_exclude_files = ["search.html"]
+
+# https://recommonmark.readthedocs.io/en/latest/auto_structify.html#configuring-autostructify
+def setup(app):
+    app.add_config_value('recommonmark_config', {
+            'enable_auto_toc_tree': True,
+            'auto_toc_maxdepth': 2,
```

Comment:
I suggest changing this to three levels of headings. Note: do not make it deeper than three though. The rule of thumb is: if at any point a need arises for a head4, consider refactoring the TOC globally instead. Summary: don't allow hierarchy deeper than 3 levels. Reasons: anyting deeper and users  get tired, confused, and content stops being easily discoverable. (pain ensues)

Reply:
@JuliaMalkin this configures how many levels are automatically displayed when rendering a table of contents.

E.g.

<img width="621" alt="Screen Shot 2019-10-03 at 6 54 36 PM" src="https://user-images.githubusercontent.com/1477317/66169694-451df580-e60f-11e9-8173-8da73d5f6972.png">

^the table of contents only gets **rendered** for 2 levels, even if in actuality there are more levels of headers. This is just for navigation UX. This does not limit how many levels we have in our docs hierarchy.